code/backend/batch/utilities/search/search_handler_base.py

```diff
@@ -12,10 +12,15 @@ def __init__(self, env_helper: EnvHelper):
         self.env_helper = env_helper
         self.search_client = self.create_search_client()
 
-    def search_with_facets(self, query: str, facets: list[str]):
+    def search_with_facets(self, query: str, facets: list[str], facet_count: int = None):
         if self.search_client is None:
             return None
-        return self.search_client.search(query, facets=facets)
+
+        # Construct facet parameter based on facet_count
+        facet_param = facets[0] if facet_count is None else f"{facets[0]},count:{facet_count}"
```

Comment:
I think we can hard code this to `f"{facets[0]},count:0"` to always return **all** results. I don't think we have a usecase of this function which needs to limit or will page over the results.

Can we also add some unit tests around this change? Thanks

Reply:
Sure. We will remove if else condition and will directly pass **f"{facets[0]},count:0"** to facet_params. 

We haven't found any test cases for search_handler_base.py file.

Reply:
Ah we should add it to one of the implementation classes tests file e.g. https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/blob/main/code/tests/search_utilities/test_azure_search_handler.py

Reply:
Added test cases, can you please review it once.